models/intermediate/int_iterable__campaign_event_metrics.sql

```diff
@@ -1,30 +1,43 @@
 with user_campaign as (
-
     select *
     from {{ ref('iterable__user_campaign') }}
-
 ), campaign_user_event_metrics as (
-
-{%- set user_campaign_columns = adapter.get_columns_in_relation(ref('iterable__user_campaign')) %}
-
+    {%- set user_campaign_columns = adapter.get_columns_in_relation(ref('iterable__user_campaign')) %}
+    
     select
         campaign_id,
-        template_id,
-        experiment_id,
-        count(distinct unique_user_key) as count_unique_users
+        template_id
+        
+        {%- if var('iterable__using_event_extension', True) %}
+        , experiment_id
+        {%- endif %}
+        
+        , count(distinct unique_user_key) as count_unique_users
+        
         {% for col in user_campaign_columns %}
-            {% if col.name|lower not in ['unique_user_key', 'user_id', '_fivetran_user_id', 'user_email', 'user_full_name', 'campaign_id', 'campaign_name', 'recurring_campaign_id', 
-                                        'recurring_campaign_name', 'first_event_at', 'last_event_at', 'template_id', 'template_name',
-                                        'experiment_id'] %}
-            , sum( {{ col.name }} ) as {{ col.name }}
-            , count(distinct case when {{ col.name }} > 0 then user_email else null end) as {{ 'unique_' ~ col.name }}
+            {% if col.name|lower not in ['unique_user_key', 'user_id', '_fivetran_user_id', 'user_email', 'user_full_name', 'campaign_id', 'campaign_name', 'recurring_campaign_id', 'recurring_campaign_name', 'first_event_at', 'last_event_at', 'template_id', 'template_name'] %}
+                {% if col.name|lower == 'experiment_id' %}
+                    {% if col.name in user_campaign_columns %}
+                        {% if var('iterable__using_event_extension', True) %}
+                            , sum({{ col.name }}) as {{ col.name }}
```

Comment:
@fivetran-reneeli 

Two questions:  
1) Is there a reason we are summing `experiment_id`? (experiment ids usually don't have much aggregate value in my experience). The count logic makes sense. I think adding details around both new columns to the CHANGELOG would be helpful, since it looks as if they are being brought into end models.

2) `experiment_id` would be a duplicate column if this logic compiled (since `iterable__using_event_extension` being True holds for both cases when `experiment_id` is brought in by line 12). I think this column value needs to be changed because if it compiles it will lead to duplicate errors.

3) These changes will impact `iterable__campaigns`. I'm wondering if we should also write a similar consistency test for that model as well to be on the safe side.

Reply:
Thanks Avinash for noticing this, `experiment_id` was not meant to be part of the aggregations and I see that the logic I added didn't exclude for it when `iterable__using_event_extension` is True. It looks like just keeping it in the exclusion list does the trick.